src/pages/main.tsx

```diff
@@ -30,26 +31,33 @@ import NodeSelector from '../components/NodeSelectorPanel';
 import ReactFlowBox from '../components/ReactFlowBox';
 import { GlobalProvider } from '../helpers/contexts/GlobalNodeState';
 import { SettingsProvider } from '../helpers/contexts/SettingsContext';
+import { BackendNodesResponse } from '../helpers/Backend';
+import { SchemaMap } from '../common-types';
 
-const Main = ({ port }) => {
+interface MainProps {
+  port: number;
+}
+
+const Main = ({ port }: MainProps) => {
   // console.log('🚀 ~ file: main.jsx ~ line 27 ~ Main ~ port', port);
-  const [availableNodes, setAvailableNodes] = useState(null);
-  const [nodeTypes, setNodeTypes] = useState(null);
-  const edgeTypes = {
+  const [availableNodes, setAvailableNodes] = useState<SchemaMap | null>(null);
+  const [nodeTypes, setNodeTypes] = useState<NodeTypes | null>(null);
+  const edgeTypes: EdgeTypes = {
     main: CustomEdge,
   };
   // const { colorMode, toggleColorMode } = useColorMode();
   const [, height] = useWindowSize();
 
-  const reactFlowWrapper = useRef(null);
+  const reactFlowWrapper = useRef<HTMLDivElement>(null);
 
   // Queries
   const [backendReady, setBackendReady] = useState(false);
 
-  const options = { cachePolicy: 'no-cache', retries: 10 };
-  const { loading, error, data, response } = useFetch(`http://localhost:${port}/nodes`, options, [
-    port,
-  ]);
+  const { loading, error, data, response } = useFetch<BackendNodesResponse>(
+    `http://localhost:${port}/nodes`,
```

Comment:
Didn't you find a different module for making requests? Shouldn't we just use that everywhere?

Reply:
Yes, but also no. The backend class I made returns promises, and awaiting those in components isn't trivial. It's not hard either, but the main goal of this PR was to port those files over.